src/js/server/models/Hacker.ts

```diff
@@ -95,6 +95,60 @@ export function getTicketStatus(hackerApplication: HackerApplicationInstance) {
   });
 }
 
+// Returns a promise that resolves to the headline application status
+async function deriveOverallStatus() {
+  const [applicationStatus, teamApplicationStatus, responseStatus, rsvpStatus, ticketStatus] = await Promise.all([
+    this.getApplicationStatus(), this.getTeamApplicationStatus(), this.getResponseStatus(), this.getRsvpStatus(), this.getTicketStatus()
+  ]);
+  if (applicationStatus == statuses.application.INCOMPLETE || teamApplicationStatus == statuses.application.INCOMPLETE)
+    return process.env.APPLICATIONS_OPEN_STATUS === statuses.applicationsOpen.OPEN ? statuses.overall.INCOMPLETE : statuses.overall.INCOMPLETE_CLOSED;
+  else if (responseStatus == statuses.response.PENDING)
+    return statuses.overall.IN_REVIEW;
+  else if (responseStatus == statuses.response.REJECTED)
+    return statuses.overall.REJECTED;
+  else if (ticketStatus == statuses.ticket.HAS_TICKET)
+    return statuses.overall.HAS_TICKET;
+  else if (rsvpStatus == statuses.rsvp.INCOMPLETE)
+    return statuses.overall.INVITED_AWAITING_RSVP;
+  else if (rsvpStatus == statuses.rsvp.COMPLETE_NO)
+    return statuses.overall.INVITED_DECLINED;
+  else if (rsvpStatus == statuses.rsvp.COMPLETE_EXPIRED)
+    return statuses.overall.INVITED_EXPIRED;
+  else if (rsvpStatus == statuses.rsvp.COMPLETE_YES)
+    return statuses.overall.INVITED_ACCEPTED;
+  else {
+    console.log('Couldn\'t derive an overall status');
+    console.log({
+      applicationStatus,
+      responseStatus,
+      teamApplicationStatus,
+      rsvpStatus,
+      ticketStatus,
+    });
+    throw new Error('Couldn\'t derive an overall status');
+  }
+};
+
+async function getStatuses() {
+  const statuses = {};
+  [ 
+    statuses['applicationStatus'],
+    statuses['teamApplicationStatus'],
+    statuses['responseStatus'],
+    statuses['rsvpStatus'],
+    statuses['ticketStatus'],
+    statuses['overallStatus']
+  ] = await Promise.all([
+    this.getApplicationStatus(),
+    this.getTeamApplicationStatus(),
+    this.getResponseStatus(),
+    this.getRsvpStatus(),
+    this.getTicketStatus(),
+    this.deriveOverallStatus(),
```

Comment:
I like how we've collected the statuses together, however I think it might be nice to make a TypeScript interface for the object and use syntax like `const statuses: HackerStatuses = { applicationStatus: ..., teamApplicationStatus: ..., ... }`.  We would then get type checking on statuses.